### PR TITLE
[Fuzzing] add  cifuzz workflows

### PR DESCRIPTION
**Pull request by 0x34d**
Add [CIFuzz](https://google.github.io/oss-fuzz/getting-started/continuous-integration) for PR fuzzing.
This will fuzz the PR to find bugs in it.

In the future, if CIFuzz or oss-fuzz fail, the main reason could be https://github.com/google/oss-fuzz/pull/10729/files#diff-11e8ba3b20d92c1a488036e14076d4b1bf96313224427f03a8df69d187205247R21 line.

Update for `DCO test`: I used the GitHub's built-in editor for this PR.

### CODE REVIEW

**Comment by pmazzini** [comment]
Please sign the commit. Example:

* git rebase HEAD~1 --signoff
* git push --force-with-lease origin cifuzz.yml

**Review by pmazzini — APPROVED** [review]

File: .github/workflows/cifuzz.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: CIFuzz
+on: [pull_request]
+jobs:
+  Fuzzing:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Build Fuzzers
+      id: build
+      uses: google/oss-fuzz/infra/cifuzz/actions/build_fuzzers@master
```
Comment (pmazzini) [inline .github/workflows/cifuzz.yml:9]: Should we be using the master branch?

Comment (0x34d) [reply .github/workflows/cifuzz.yml:9]: Default branch is master so ya.

Comment (0x34d) [reply .github/workflows/cifuzz.yml:9]: link : https://google.github.io/oss-fuzz/getting-started/continuous-integration/#branches-and-paths

Comment (0x34d) [reply .github/workflows/cifuzz.yml:9]: >       uses: google/oss-fuzz/infra/cifuzz/actions/build_fuzzers@master

is a docker path

File: .github/workflows/cifuzz.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+permissions: {}
+jobs:
+ Fuzzing:
+   runs-on: ubuntu-latest
+   permissions:
+     security-events: write
+   steps:
+   - name: Build Fuzzers
+     id: build
+     uses: google/oss-fuzz/infra/cifuzz/actions/build_fuzzers@master
```
Comment (0x34d) [inline .github/workflows/cifuzz.yml:12]: This is a Docker file used to build the project.

something like this:
```bash
python infra/helper.py build_image   $PROJECT_NAME
python infra/helper.py build_fuzzers $PROJECT_NAME --engine libfuzzer --sanitizer address
python infra/helper.py check_build   $PROJECT_NAME
```

File: .github/workflows/cifuzz.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+     with:
+       oss-fuzz-project-name: 'go-dhcp'
+       language: go
+   - name: Run Fuzzers
+     uses: google/oss-fuzz/infra/cifuzz/actions/run_fuzzers@master
```
Comment (0x34d) [inline .github/workflows/cifuzz.yml:17]: This is a Docker file use for running the fuzzers:
``` bash
python infra/helper.py run_fuzzer   $PROJECT_NAME --corpus-dir=<path-to-temp-corpus-dir>  <fuzz_target>
```

**Comment by 0x34d** [comment]
@pmazzini ping.

**Review by hugelgupf — APPROVED** [review]

**Pull request merged**